cmd/neofs-node/config.go

```diff
@@ -367,7 +367,9 @@ func initLocalStorage(c *cfg) {
 func initShardOptions(c *cfg) {
 	var opts [][]shard.Option
 
-	engineconfig.IterateShards(c.appCfg, func(sc *shardconfig.Config) {
+	require := !nodeconfig.Relay(c.appCfg) // relay node does not require shards
+
+	engineconfig.IterateShards(c.appCfg, require, func(sc *shardconfig.Config) {
```

Comment:
Wy don't we just skip the whole iteration?

Reply:
For now there is no need to completely skip the section. Perhaps this will change with a more detailed study of the relay mode.